sdk/core/azure-core-serializer-avro-apache/src/main/java/com/azure/core/serializer/avro/apache/ApacheAvroSerializer.java

```diff
@@ -41,26 +44,49 @@
         this.specificData = specificData;
     }
 
+    @Override
+    public <T> T deserialize(byte[] data, TypeReference<T> typeReference) {
+        return deserialize(data, () -> decoderFactory.binaryDecoder(data, null));
+    }
+
     @Override
     public <T> T deserialize(InputStream stream, TypeReference<T> typeReference) {
-        if (stream == null) {
+        return deserialize(stream, () -> decoderFactory.binaryDecoder(stream, null));
+    }
+
+    private <T> T deserialize(Object data, Supplier<Decoder> decoderSupplier) {
+        if (data == null) {
             return null;
         }
 
         DatumReader<T> reader = new SpecificDatumReader<>(schema, schema, specificData);
 
         try {
-            return reader.read(null, decoderFactory.binaryDecoder(stream, null));
+            return reader.read(null, decoderSupplier.get());
         } catch (IOException ex) {
             throw logger.logExceptionAsError(new UncheckedIOException(ex));
         }
     }
 
+    @Override
+    public <T> Mono<T> deserializeAsync(byte[] data, TypeReference<T> typeReference) {
+        return Mono.fromCallable(() -> deserialize(data, typeReference));
+    }
+
     @Override
     public <T> Mono<T> deserializeAsync(InputStream stream, TypeReference<T> typeReference) {
         return Mono.fromCallable(() -> deserialize(stream, typeReference));
     }
 
+    @Override
+    public byte[] serialize(Object value) {
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+
+        serialize(stream, value);
+
+        return stream.toByteArray();
+    }
```

Comment:
Same for concrete implementations too - is there a need to override default implementation in `ObjectSerializer`?

Reply:
Given this is functionally equivalent there is no need to implement it, if in the future an API was offered that directly returns `byte[]` we can implement it.